generator/lib/src/field_element_ex.dart

```diff
@@ -137,27 +137,58 @@ extension FieldElementEx on FieldElement {
+          final file = span!.file;
+          final listOfValidTypes = RealmPropertyType.values //
+              .map((t) => t.mapping)
+              .where((m) => m.canBePrimaryKey)
+              .map((m) => m.type);
+
+          throw RealmInvalidGenerationSourceError(
+            'Realm only supports the @PrimaryKey annotation on fields of type\n${listOfValidTypes.join(', ')}\nas well as their nullable versions',
+            element: this,
+            primarySpan: typeSpan(file),
+            primaryLabel: "$modelTypeName is not a valid type here",
+            todo: "Change the type of '$displayName' or remove the @PrimaryKey annotation",
+          );
+        }
       }
 
-      // Validate indexes
-      if ((((primaryKey ?? indexed) != null) && !(type.realmType?.mapping.indexable ?? false)) || //
-          (primaryKey != null && (type.isDartCoreBool || type.isExactly<RealmValue>()))) {
+      final indexType = RealmIndexType.values.elementAt(indexed?.value.getField("indexType")!.getField("index")!.toIntValue() ?? RealmIndexType.none.index);
+
+      if (indexed != null) {
         final file = span!.file;
-        final annotation = (primaryKey ?? indexed)!.annotation;
-        final listOfValidTypes = RealmPropertyType.values //
-            .map((t) => t.mapping)
-            .where((m) => m.indexable && ((m.type != bool && m.type != RealmValue) || primaryKey == null))
-            .map((m) => m.type);
+        if (indexType == RealmIndexType.none) {
+          throw RealmInvalidGenerationSourceError("Don't use RealmIndexType.none for @Indexed annotations",
+              element: this,
+              primarySpan: typeSpan(file),
+              primaryLabel: '@Indexed(RealmIndexType.none) is not valid annotation here',
+              todo: 'Remove the @Indexed annotation or change the index type to a value other than none');
+        }
 
-        throw RealmInvalidGenerationSourceError(
-          'Realm only supports the $annotation annotation on fields of type\n${listOfValidTypes.join(', ')}\nas well as their nullable versions',
-          element: this,
-          primarySpan: typeSpan(file),
-          primaryLabel: "$modelTypeName is not a valid type here",
-          todo: //
-              "Change the type of '$displayName', "
-              "or remove the $annotation annotation",
-        );
+        if (indexType == RealmIndexType.fullText && type.realmType != RealmPropertyType.string) {
+          throw RealmInvalidGenerationSourceError('Cannot add full-text index on a non-string property',
+              element: this,
+              primarySpan: typeSpan(file),
+              primaryLabel: 'Cannot use RealmIndexType.fullText for property of type $modelTypeName',
+              todo: 'Change the index type to general or change the property type to string');
+        }
+
+        if (type.realmType?.mapping.indexable != true) {
+          final listOfValidTypes = RealmPropertyType.values //
```

Comment:
```suggestion
          final listOfValidTypes = RealmPropertyType.values
```
for me the formatter does not require these `//` at the end

Reply:
Same comment as above - it's moved code and removing the slashes moves everything on a single line.

Reply:
lets leave it. for me it does not move.